packages/gestalt/src/TapLink.js

```diff
@@ -0,0 +1,112 @@
+// @flow strict
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import touchableStyles from './Touchable.css';
+import styles from './TapLink.css';
+import useTapFeedback, { keyPressShouldTriggerTap } from './useTapFeedback.js';
+import getRoundingClassName, {
+  RoundingPropType,
+  type Rounding,
+} from './getRoundingClassName.js';
+
+type TapEvent =
+  | SyntheticMouseEvent<HTMLAnchorElement>
+  | SyntheticKeyboardEvent<HTMLAnchorElement>;
+
+type Props = {|
+  children?: React.Node,
+  hoverStyle?: 'none' | 'underline',
+  href: string,
+  inline?: boolean,
+  onTap?: ({ event: TapEvent }) => void,
+  rel?: 'none' | 'nofollow',
+  rounding?: Rounding,
+  tapStyle?: 'none' | 'compress',
+  target?: null | 'self' | 'blank',
+|};
+
+function TapLink({
+  children,
+  href,
+  inline = false,
+  onTap,
+  rel = 'none',
+  rounding = 0,
+  hoverStyle = 'none',
+  tapStyle = 'none',
+  target = null,
+}: Props) {
+  const {
+    isTapping,
+    handleBlur,
+    handleMouseDown,
+    handleMouseUp,
+    handleTouchStart,
+    handleTouchMove,
+    handleTouchCancel,
+    handleTouchEnd,
+  } = useTapFeedback();
+
+  const className = classnames(
+    touchableStyles.touchable,
+    styles.tapLink,
+    inline ? styles.inlineBlock : styles.block,
+    getRoundingClassName(rounding),
+    {
+      [styles.hoverUnderline]: hoverStyle === 'underline',
+      [touchableStyles.tapCompress]: tapStyle === 'compress' && isTapping,
+    }
+  );
+
+  return (
+    <a
+      className={className}
+      href={href}
+      onContextMenu={event => event.preventDefault()}
+      onClick={event => {
+        if (onTap) {
+          onTap({ event });
+        }
+      }}
+      onBlur={handleBlur}
+      onMouseDown={handleMouseDown}
+      onMouseUp={handleMouseUp}
+      onKeyPress={event => {
+        // Check to see if space or enter were pressed
+        if (onTap && keyPressShouldTriggerTap(event)) {
+          // Prevent the default action to stop scrolling when space is pressed
+          event.preventDefault();
+          onTap({ event });
+        }
+      }}
+      onTouchStart={handleTouchStart}
+      onTouchMove={handleTouchMove}
+      onTouchCancel={handleTouchCancel}
+      onTouchEnd={handleTouchEnd}
+      rel={[
+        ...(target === 'blank' ? ['noopener', 'noreferrer'] : []),
```

Comment:
@jackhsu978 I like this a lot - we should follow the same logic in `<Link />` (we should do that in a follow up PR)

Reply:
sure